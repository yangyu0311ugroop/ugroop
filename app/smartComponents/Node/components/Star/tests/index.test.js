import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { Star, toggleStar } from '../index';

describe('toggleStar()', () => {
  it('should star', () => {
    expect(toggleStar(1123)()).toEqual([1123]);
  });

  it('should unstar', () => {
    expect(toggleStar(1123)([1, 1123])).toEqual([1]);
  });
});

describe('<Star />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    id: 1122,
    classes: {},
    resaga,
  };

  beforeEach(() => {
    rendered = shallow(<Star {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(Star).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('toggleStar()', () => {
    it('should toggleStar()', () => {
      instance.toggleStar({ stopPropagation: jest.fn() });

      expect(resaga.setValue).toBeCalled();
    });
  });

  describe('render()', () => {
    it('should render children', () => {
      const children = jest.fn(() => 'children');
      rendered.setProps({ children });

      expect(instance.render()).toBe('children');
    });

    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
