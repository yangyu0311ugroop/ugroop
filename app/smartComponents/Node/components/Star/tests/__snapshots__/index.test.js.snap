// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star /> render() should render correctly 1`] = `
<div>
  <WithStyles(Button)
    buttonTitle="Click to star or unstar. Starred items show up at the top of your list."
    className=""
    dense={true}
    icon="lnr-star"
    iconButton={true}
    onClick={[Function]}
    size="extraSmall"
  />
</div>
`;
